### PR TITLE
Feature/config cache

### PR DESCRIPTION
**Pull request by hlgp**

### CODE REVIEW

File: warehouse/index-stats/src/test/java/datawave/mapreduce/shardStats/StatsHyperLogMapperTest.java [hunk 1]
```diff
@@ -85,6 +90,17 @@ private void runDriver(List<StatsTestData> entries) throws IOException, Interrup
         conf.set(StatsHyperLogMapper.STATS_MAPPER_OUTPUT_INTERVAL, "4");
         conf.set(StatsHyperLogMapper.STATS_MAPPER_UNIQUE_COUNT, "true");
         conf.set(StatsJob.STATS_VISIBILITY, "vis");
+        conf.set("accumulo.username", "user");
+        conf.set("accumulo.password", "user");
+        conf.set("accumulo.instance.name", "user");
```
Comment (ivakegg) [inline warehouse/index-stats/src/test/java/datawave/mapreduce/shardStats/StatsHyperLogMapperTest.java:95]: this "user" gets around...

File: warehouse/ingest-core/src/main/java/datawave/ingest/config/BaseHdfsFileCacheUtil.java [hunk 2]
```diff
@@ -17,7 +17,8 @@ public abstract class BaseHdfsFileCacheUtil {
     protected Path cacheFilePath;
     protected final Configuration conf;
     protected AccumuloHelper accumuloHelper;
-    private static String delimiter = "\t";
+    protected String delimiter = "\t";
+    protected short cacheReplicas = 3;
```
Comment (ivakegg) [inline warehouse/ingest-core/src/main/java/datawave/ingest/config/BaseHdfsFileCacheUtil.java:21]: cacheReplicas only used in TableConfigCache.  Perhaps it should be moved there

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java [hunk 3]
```diff
@@ -296,7 +295,8 @@ public int run(String[] args) throws Exception {
         }
         
         TableConfigurationUtil tableConfigUtil = new TableConfigurationUtil(conf);
-        tableNames = tableConfigUtil.getTableNames();
+        tableConfigUtil.registerTableNamesFromConfigFiles(conf);
+        tableNames = tableConfigUtil.getJobOutputTableNames(conf);
```
Comment (ivakegg) [inline warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java:299]: nice

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducer.java [hunk 4]
```diff
@@ -57,13 +55,16 @@
     
     public static Text ALL_COLUMN_FAMILIES = new Text("*");
     
+    public static boolean alreadySetup = false;
```
Comment (ivakegg) [inline warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducer.java]: not used

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableConfigurationUtil.java [hunk 5]
```diff
@@ -20,59 +19,111 @@
...
 import java.util.Set;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+/**
+ * This class serves as the liaison between datawave job configuration and accumulo tables. Most of this was ripped out of IngestJob for more convenient reuse
+ * in other Jobs
+ **/
 public class TableConfigurationUtil {
```
Comment (ivakegg) [inline warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableConfigurationUtil.java:52]: To we have any unit test cases for this class?

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/index-stats/src/test/java/datawave/mapreduce/shardStats/StatsHyperLogMapperTest.java (hunk 1 above)
Comment (hlgp) [reply warehouse/index-stats/src/test/java/datawave/mapreduce/shardStats/StatsHyperLogMapperTest.java:95]: indeed

File: warehouse/ingest-core/src/main/java/datawave/ingest/config/BaseHdfsFileCacheUtil.java (hunk 2 above)
Comment (hlgp) [reply warehouse/ingest-core/src/main/java/datawave/ingest/config/BaseHdfsFileCacheUtil.java:21]: I figured we would want to use this for other types of HDFS file caches in future, but wasn't sure if implementing it in all of them was too much scope creep.

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducer.java (hunk 4 above)
Comment (hlgp) [reply warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducer.java]: Will remove

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableConfigurationUtil.java (hunk 5 above)
Comment (hlgp) [reply warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableConfigurationUtil.java:52]: Parts of it are tested in AggregatingReducerTest and BulkIngestKeyAggregatingReducerTest, but I should be more thorough than that.

**Review by ivakegg — APPROVED** [review]

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducerTest.java [hunk 6]
```diff
@@ -30,14 +32,18 @@
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.PowerMockRunner;
 
+import static datawave.ingest.mapreduce.job.TableConfigurationUtil.ITERATOR_CLASS_MARKER;
+
 @RunWith(PowerMockRunner.class)
-@PrepareForTest({TableConfigurationUtil.class})
+// @PrepareForTest({TableConfigurationUtil.class})
```
Comment (apmoriarty) [inline warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducerTest.java:38]: remove comment

File: warehouse/ingest-csv/src/test/resources/config/ingest/metadata-config2.xml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+<property>
+	<name>metadata.handler.classes</name>
+	<value></value>
+</property>
+
+<property>
+  <name>metadata.table.name</name>
+  <value>datawave.metadata</value>
+</property>
```
Comment (apmoriarty) [inline warehouse/ingest-csv/src/test/resources/config/ingest/metadata-config2.xml:22]: let's pick a consistent indent

**Review by apmoriarty — CHANGES_REQUESTED** [review]

**Review by apmoriarty — APPROVED** [review]

**Pull request merged**